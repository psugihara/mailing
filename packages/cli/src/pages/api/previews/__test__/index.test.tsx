import { NextApiRequest, NextApiResponse } from "next";
import { createMocks } from "node-mocks-http";
import index from "..";

describe("index", () => {
  it("should return previews", async () => {
    const { req, res } = createMocks({
      method: "GET",
    });

    await index(
      req as unknown as NextApiRequest,
      res as unknown as NextApiResponse
    );

    expect(res.statusCode).toBe(200);
    const json = res._getJSONData();
    expect(json).toMatchSnapshot();
  });

  it("should return template subject if it exists", async () => {
    const { req, res } = createMocks({
      method: "GET",
    });

    await index(
      req as unknown as NextApiRequest,
      res as unknown as NextApiResponse
    );

    expect(res.statusCode).toBe(200);
    const json = res._getJSONData();
    expect(json["previewText"]["/previews/AccountCreated/accountCreated"]).toBe(
      "Welcome to BookBook, Amelita!"
    );
  });
});
