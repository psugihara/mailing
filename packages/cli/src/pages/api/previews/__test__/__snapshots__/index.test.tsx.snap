// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index should return previews 1`] = `
{
  "previewText": {
    "/previews/AccountCreated/accountCreated": "Welcome to BookBook, Amelita!",
    "/previews/NewSignIn/newSignIn": "Security Alert: New Sign-In Hello Amelita, We noticed a new sign-in to your BookBook account on a Mac device. If this was you, you don’t nee",
    "/previews/Reservation/reservationChanged": "Reservation Changed • Salazar in Silver Lake • Sunday, Aug 22 at 1:30pm • Party of 4, patio You’re all set! Your reservation at Salazar has ",
    "/previews/Reservation/reservationConfirmed": "Reservation Confirmed • Salazar in Silver Lake • Saturday, Aug 22 at 1:30pm • Party of 4, patio Thanks for booking your reservation at Salaz",
    "/previews/Reservation/reservationWithError": "Reservation Canceled • Salazar in Silver Lake • Sunday, Aug 22 at 1:30pm • Party of 4, patio If this was a mistake or if you changed your mi",
    "/previews/ResetPassword/resetPassword": "Hello Amelita, We’ve received your request to change your password. Use the link below to set up a new password for your account. This link ",
  },
  "previews": [
    [
      "AccountCreated",
      [
        "accountCreated",
      ],
    ],
    [
      "NewSignIn",
      [
        "newSignIn",
      ],
    ],
    [
      "Reservation",
      [
        "reservationWithError",
        "reservationConfirmed",
        "reservationChanged",
      ],
    ],
    [
      "ResetPassword",
      [
        "resetPassword",
      ],
    ],
  ],
}
`;
